source/neuropod/multiprocess/shm_tensor.hh

```diff
@@ -192,9 +192,10 @@ public:
                       const SHMBlockID &          block_id)
         : TypedNeuropodTensor<std::string>(copy_and_strip_last_dim(dims)), write_buffer_(this->get_num_elements())
     {
-        auto base_ptr =
-            stdx::make_unique<SHMNeuropodTensor<uint8_t>>(dims, std::move(block), data, block_id)->get_raw_data_ptr();
-        auto max_len = dims[dims.size() - 1];
+        auto byte_block = stdx::make_unique<SHMNeuropodTensor<uint8_t>>(dims, std::move(block), data, block_id);
+
+        auto base_ptr = byte_block->get_raw_data_ptr();
+        auto max_len  = dims[dims.size() - 1];
 
         // Copy into our local buffer
         auto numel = get_num_elements();
```

Comment:
btw, just found, this is not related to you change but anyway:
1) It would be more consistent to init numel with write_buffer_.size() here (or just use write_buffer_.size() directly):
auto numel = write_buffer_.size().

2) I found that then for loop looks like this:
```
// Copy into our local buffer
        auto numel = get_num_elements();
        for (int i = 0; i < numel; i++)
        {
            auto pos     = base_ptr + i * max_len;
```
You see, numel is size_t here, iterator i is int. Theoretically this will cause infinite loop on 64 bit machine where int is 32bit and numel is greater than it can fit.

I saw some other places where for(int i =) was used, I am surprised that static check/linter doesn't report it. I understand that this is not critical on current scale but it would be nice to have it consistent with 64bit. In this case if we decide to test Neuropod's limits on a large numbers, we would not hit these obvious issues.